src/_posts/2020-05-19-promises-and-lies.md

```diff
@@ -0,0 +1,151 @@
+---
+layout: post
+asset-type: post
+name: promises-and-lies
+title: Promises and Lies
+date: 2020-05-19 10:00:00 +00:00
+author: Christopher Eyre
+description: Adding timeouts to promises in Javascript.
+image:
+    src: /assets/custom/img/blog/lie.png
+    attribution:
+        text: Public domain photo from Pixabay
+        href: https://pixabay.com/photos/lying-promises-deception-dishonesty-1562272/
+tags:
+    - javascript
+    - node
+    - reliability
+abstract: Adding timeouts to promises in Javascript. Introducing lie.js 
+alias: [/2020/05/19/promises-and-lies]
+---
+
+## Promises in Javascript
+
+Promises were added to Javascript as a standard means to avoid nesting callback functions.
+Before promises each function could have it's own functions that would be called once the job is completed. They are mostly used for asynchronous events. They are not a means of lazy evaluation or for process control. Functions that have an async result typically now return a Promise.
+
+Promises have the following format:
+
+```javascript
+var promiseObj = new Promise(executor);
```

Comment:
Do you feel particularly strongly about `promiseObj`. I'm thinking of Uncle Bob in Clean Code. On naming he say you should be able to pronounce variable names easily and he suggests not to include the type in the name. I'm with him on both these points.

```suggestion
var promise = new Promise(executor);
```